src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,459 @@
+        {
+            foreach (var platformSpecificOperation in platformSpecificOperations)
+            {
+                ReportDiagnostics(platformSpecificOperation.Key, platformSpecificOperation.Value.FirstOrDefault(), context);
+            }
+        }
+
+        private static void ReportDiagnostics(IOperation operation, PlatformAttributeInfo attribute, OperationBlockAnalysisContext context)
+        {
+            string operationName = string.Empty;
+
+            if (operation is IInvocationOperation iOperation)
+            {
+                operationName = iOperation.TargetMethod.Name;
+            }
+            else if (operation is IPropertyReferenceOperation pOperation)
+            {
+                operationName = pOperation.Property.Name;
+            }
+            else if (operation is IFieldReferenceOperation fOperation)
+            {
+                operationName = fOperation.Field.Name;
+            }
+
+            context.ReportDiagnostic(operation.CreateDiagnostic(SwitchRule(attribute.AttributeType), operationName, attribute.OsPlatformName, attribute.Version.ToString()));
+        }
+
+        private static void AnalyzeInvocationOperation(IOperation operation, OperationAnalysisContext context,
+            List<PlatformAttributeInfo>? parsedTfms, ref PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>> platformSpecificOperations)
+        {
+            using var builder = ArrayBuilder<PlatformAttributeInfo>.GetInstance();
+            AttributeData[]? attributes = null;
+
+            if (operation is IInvocationOperation iOperation)
+            {
+                attributes = FindAllPlatformAttributesApplied(iOperation.TargetMethod.GetAttributes(), iOperation.TargetMethod.ContainingType);
+            }
+            else if (operation is IPropertyReferenceOperation pOperation)
+            {
+                if (!IsWithinConditionalOperation(operation))
+                    attributes = FindAllPlatformAttributesApplied(pOperation.Property.GetAttributes(), pOperation.Property.ContainingType);
+            }
+            else if (operation is IFieldReferenceOperation fOperation)
+            {
+                if (!IsWithinConditionalOperation(operation))
+                    attributes = FindAllPlatformAttributesApplied(fOperation.Field.GetAttributes(), fOperation.Field.ContainingType);
+            }
+
+            if (attributes != null)
+            {
+                foreach (AttributeData attribute in attributes)
+                {
+                    if (PlatformAttributeInfo.TryParseAttributeData(attribute, out PlatformAttributeInfo parsedAttribute))
+                    {
+                        if (!(IsSuppressedByTfm(parsedTfms, parsedAttribute) || IsSuppressedByAttribute(parsedAttribute, context.ContainingSymbol)))
```

Comment:
As said elsewhere, I believe we should remove `IsSuppressedByTfm(parsedTfms, parsedAttribute)` because MSBuild will emit an assembly level attributed based on the TFM already. This keeps the logic of TFM outside the analyzer.